api/v1alpha1/conditions.go

```diff
@@ -100,11 +100,17 @@ func FindStatusCondition(conditions []Condition, conditionType ConditionType) *C
 
 // IsStatusConditionTrue returns true when the conditionType is present and set to `ConditionTrue`
 func IsStatusConditionTrue(conditions []Condition, conditionType ConditionType) bool {
+	if len(conditions) == 0 {
+		return false
+	}
 	return IsStatusConditionPresentAndEqual(conditions, conditionType, True)
 }
 
 // IsStatusConditionFalse returns true when the conditionType is present and set to `ConditionFalse`
 func IsStatusConditionFalse(conditions []Condition, conditionType ConditionType) bool {
+	if len(conditions) == 0 {
+		return true
+	}
```

Comment:
I understand your motivation here, but technically, if a Condition isn't set, its value should be assumed to be "Unknown."

Reply:
Fixed it in the task. Keeping these functions untouched.